application/controllers/Home.php

```diff
@@ -36,29 +37,63 @@ public function index()
 	}
 
 	public function do_upload()
-	{
+	{ 
 		$config['upload_path']          = './uploads/';
-		$config['allowed_types']        = 'gif|jpg|png';
+		$config['allowed_types']        = 'pdf|jpg|png';
 		$config['max_size']             = 1024;
 		// $config['max_width']            = 2048;
 		// $config['max_height']           = 2048;
-		$data['title'] = 'Home';
+		$data['tab'] = 'Oops!';
+		$data['title'] = 'Ha habido un error';
 
 		$this->load->library('upload', $config);
 
-		if ( ! $this->upload->do_upload('userfile'))
+		$filesError = array(0,0); //0 = Error, 1 = Success
+
+		if ( ! $this->upload->do_upload('userfileplan'))
```

Comment:
If innecesario, la funcion $this->upload->do_upload retorna falso si no lo sube. Ademas estamos intentando subir el archivo sin nisiquiera verificar que hay un archivo para subir. Ahí hay unos puntos a mejorar.

Reply:
Enterado, aunque ocuparía una descripción mas elaborada porque creo que me estoy confundiendo con esto.